Source/Core/AudioCommon/WASAPIStream.cpp

```diff
@@ -39,177 +45,123 @@ WASAPIStream::WASAPIStream()
 
 WASAPIStream::~WASAPIStream()
 {
-  if (m_enumerator)
-    m_enumerator->Release();
-
-  if (m_need_data_event)
-    CloseHandle(m_need_data_event);
-
-  if (m_running)
-  {
-    m_running = false;
-    if (m_thread.joinable())
-      m_thread.join();
-  }
-
-  CoUninitialize();
+  m_running.store(false, std::memory_order_relaxed);
+  if (m_thread.joinable())
+    m_thread.join();
 }
 
 bool WASAPIStream::isValid()
 {
   return true;
 }
 
-static bool HandleWinAPI(std::string message, HRESULT result)
+static bool HandleWinAPI(std::string_view message, HRESULT result)
 {
-  if (result != S_OK)
+  if (FAILED(result))
   {
-    _com_error err(result);
-    std::string error = TStrToUTF8(err.ErrorMessage()).c_str();
+    std::string error;
```

Comment:
minor optimization ([saves like 10 lines of assembly](https://godbolt.org/z/Lf3-u1)), but you can avoid reassigning a string by using an immediately invoked lambda 

```suggestion
    const std::string error = [&result] {
       switch (result)
       {
       case AUDCLNT_E_DEVICE_IN_USE:
           return "Audio endpoint already in use!";
       default:
           return TStrToUTF8(_com_error(result).ErrorMessage()).c_str();
       }
    }();
```

Reply:
Eh, not a fan. This is a very cold code path (ideally is never taken) and the improvement is small enough not to warrant this change which IMO decreases readability.

It's a neat trick, but I don't think this is the place to use it.

Reply:
totally fair, just a big fan of immediately invoked lambdas :)

Reply:
Sure thing, I am using them too, mostly for static local variable initializations - but I also think this is not the place to go for it :)